include/aos/common/array.hpp

```diff
@@ -0,0 +1,368 @@
+/*
+ * Copyright (C) 2023 Renesas Electronics Corporation.
+ * Copyright (C) 2023 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef AOS_ARRAY_HPP_
+#define AOS_ARRAY_HPP_
+
+#include <assert.h>
+
+#include "aos/common/buffer.hpp"
+#include "aos/common/error.hpp"
+
+namespace aos {
+
+/**
+ * Array instance.
+ * @tparam T array type.
+ * @tparam cMaxSize max array size.
+ */
+template <typename T, size_t cMaxSize = 0>
+class Array {
+public:
+    /**
+     * Crates array instance over the buffer.
+     *
+     * @param buffer underlying buffer.
+     * @param size current array size.
+     */
+    explicit Array(const Buffer& buffer, size_t size = 0)
+        : mSize(size)
+    {
+        SetBuffer(buffer);
+    }
+
+    /**
+     * Creates array instance from const C array.
+     *
+     * @param items const C array.
+     */
+    explicit Array(const T* items)
```

Comment:
maybe better to use range from start to end, and in this case it will be more safety ?
```
Array(const T* start, const T* end)
```
and calculate correct mSize

Reply:
You have the following constructor: `Array(const T* items)`. In your example, you used `Array<const int> array(sValue, ArraySize(sValue))`, but this constructor was not called. Instead, `Array(const T* items)` was called in the example `Array<const int> array(sValue)`. Therefore, we don't have information about the size of the passed array.